baseimages/scanner/scan/context.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package scan
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+
+	"github.com/Azure/acr-builder/baseimages/scanner/util"
+	dockerbuild "github.com/docker/cli/cli/command/image/build"
+	"github.com/docker/docker/pkg/archive"
+	"github.com/docker/docker/pkg/progress"
+	"github.com/docker/docker/pkg/streamformatter"
+	"github.com/pkg/errors"
+)
+
+type dockerSourceType int
+
+// TODO: review and eliminate this.
+const (
+	dockerSourceUnknown dockerSourceType = iota
+	dockerSourceLocal
+	dockerSourceGit
+	dockerSourceArchive
+)
+
+const (
+	archiveHeaderSize = 512
+)
+
+func (s *Scanner) obtainSourceCode(
+	ctx context.Context,
+	context string,
+	dockerfile string) (workingDir string, sha string, sourceType dockerSourceType, err error) {
+	sourceType, workingDir, err = s.getContext(ctx, context, dockerfile)
+	if err != nil {
+		return workingDir, sha, sourceType, errors.Wrap(err, "failed to obtain source code")
+	}
+
+	if sourceType == dockerSourceGit {
+		sha, err = s.GetGitCommitID(ctx, workingDir)
+	}
+
+	return workingDir, sha, sourceType, err
+}
+
+func (s *Scanner) getContext(ctx context.Context, context string, dockerfile string) (dockerSourceType, string, error) {
+	isGitURL := util.IsGitURL(context)
+	isVstsURL := util.IsVstsGitURL(context)
+	isURL := util.IsURL(context)
+
+	// If the context is remote, make the destination folder to clone or untar into.
+	if isGitURL || isVstsURL || isURL {
+		if _, err := os.Stat(s.destinationFolder); os.IsNotExist(err) {
+			if innerErr := os.Mkdir(s.destinationFolder, 0700); innerErr != nil {
```

Comment:
>0700 [](start = 48, length = 4)

please comment on this FileMode value

Reply:
done, it's just standard chmod permissions